### PR TITLE
Close #1, add ability to upload files

### LINKED ISSUE #1: Allow to upload da var data file to site
Need to run on a local server to develop this locally?

### PR DESCRIPTION
**Pull request by plocket**
Closes #1. I used the files in the example folder for testing. Sorry for the unnecessary edits to wording and a bit to styles. Had trouble resisting getting those changes in. I can specifically note those if that would help you ignore them.

### CODE REVIEW

**Comment by niharikasingh** [comment]
A few comments:

1. It would be helpful to me if you just listed the other edits in this MR. That way I can map what you intended to what's in the code. Just a one sentence description for each of the changes will suffice. 
2. I noticed a minor bug. Repro steps:

- Create a JSON file with an error. I used a blank file. 
- Upload the JSON file, see error. 
- Edit the JSON file to have proper JSON. 
- Upload file again. Error doesn't disappear and upload doesn't seem to work. 

3. Very minor stylistic thing, but it would be nice to have more padding before/after this error message. 
Screen Shot 2021-06-02 at 9 03 27 PM

**Comment by plocket** [comment]
Fair request. Detailed list of changes (let me know if it's sufficient):
1. Used actual `ignore_anywhere` instead of the default `ignore_anywhere` to exclude rows.
1. Added ability to upload a list of strings to cause a row to be excluded from the table.
2. Because of that, abstracted some functionality for setting values and handling row exclusion/ignoring input errors. Just now recognized that I went overboard - the user will not be able to edit the text by hand, so will have to fix that.
2. Added ability to upload a file to set variable data.
2. Because of that, abstracted handling the error for the variable data input. Apparently not working completely yet (thanks for catching that).
3. Abstracted functionality to change a textarea's size based on its contents with `fit_textarea_to_content`, though looking at it now, I'm not sure I'm totally happy with the way it's currently handled.
3. Renamed some variables as their purpose confused me.
2. Added/edited comments.
3. Adjusted styles and some text content to, I hope, make things more clear.
4. [Simplified error display for variable data textarea input]

[Didn't quite make them one-liners. Sorry about that.]

**Comment by plocket** [comment]
Hope that's been improved, @niharikasingh

File: generate.js [hunk 1]
```diff
@@ -450,73 +454,128 @@ let update_output = function () {
...
-    } catch ( err) {
-      console.warn( err );
-      ignore_warning.classList.add('error');
-      ignore_error.innerText = err;
-      ignore_anywhere = ignore_anywhere_default_alphabetical;  // use default
+      let maybe_exclusion_list = new_value;
+      if ( typeof new_value === `string` ) { maybe_exclusion_list = JSON.parse( new_value ); }
+      if ( !Array.isArray( maybe_exclusion_list )) {
+        // Does not yet test that every item is a string
+        let error = `Error: The data is not a JSON list of strings. This tool can only use a list of strings to exclude rows in the story table.`
```
Comment (niharikasingh) [inline generate.js:529]: I'm not sure that this is working as intended... This is how I tested it: 

1. Create JSON file with contents `{"bob"}` (this is an object, not an array, so should error)
2. Upload this JSON file in the box that says "Customize the list of text that will cause a row in the table to be excluded."
3. The error message I see is `Could not load "tests.feature". SyntaxError: Unexpected token } in JSON at position 6`, which is not as helpful as the text here

**Review by niharikasingh — APPROVED** [review]

File: generate.js (hunk 1 above)
Comment (niharikasingh) [reply generate.js:529]: Ah nvm, my object was poorly constructed, duh. This works if the file I upload says `{"bob": "duck"}`. Nothing to see here.

**Pull request merged**